vnext/react-native.config.js

```diff
@@ -1,18 +1,50 @@
-// @ts-check
-require('source-map-support').install();
+/**
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT License.
+ *
+ * @ts check
+ * @format
+ */
 
 const cli = require('@react-native-windows/cli');
 
 module.exports = {
   // **** This section defined commands and options on how to provide the Windows platform to external applications
-  commands: cli.commands,
+  commands: cli.commands.map((command) => ({
+    ...command,
+    func: lazyInstallSourcemaps(command.func),
+  })),
   platforms: {
     windows: {
       linkConfig: () => null,
-      projectConfig: cli.projectConfig,
-      dependencyConfig: cli.dependencyConfig,
+      projectConfig: lazyInstallSourcemaps(cli.projectConfig),
+      dependencyConfig: lazyInstallSourcemaps(cli.dependencyConfig),
       npmPackageName: 'react-native-windows',
     },
   },
-  healthChecks: cli.getHealthChecks()
+  healthChecks: cli.getHealthChecks(),
```

Comment:
cc @acoates-ms who added the health check code. @tido64 had mentioned that the work to execute the dependencies script at module initialization time led to a measurable delay.

Reply:
@asklar I remember you fixing an issue about the dependencies script being expensive to run. Do you remember the numbers there, and what versions were impacted? It's not ideal either way to block the process on executing an external script, but it seems like we've been pretty dogmatic on that solution.

Reply:
Maybe the CLI needs an API for getting the health checks lazily, so that we do not need to do this work when loading the module.

Reply:
Ooh that's excellent. Thank you.